x-pack/plugins/features/server/feature_registry.ts

```diff
@@ -61,6 +64,110 @@ export class FeatureRegistry {
     this.esFeatures[feature.id] = featureCopy;
   }
 
+  /**
+   * Updates definitions for the registered features using configuration overrides, if any.
+   */
+  public applyOverrides(overrides: ConfigOverridesType) {
+    for (const [featureId, featureOverride] of Object.entries(overrides)) {
+      const feature = this.kibanaFeatures[featureId];
+      if (!feature) {
+        throw new Error(
+          `Cannot override feature "${featureId}" since feature with such ID is not registered.`
+        );
+      }
+
+      if (featureOverride.disabled) {
+        feature.disabled = featureOverride.disabled;
+      }
+
+      if (featureOverride.name) {
+        feature.name = featureOverride.name;
+      }
```

Comment:
__note:__ As mentioned above, I'm considering getting rid of it due to incompatibility with i18n.